client/my-sites/domains/domain-management/list/domain-row.jsx

```diff
@@ -387,9 +391,16 @@ class DomainRow extends PureComponent {
 							? translate( 'View transfer' )
 							: translate( 'View settings' ) }
 					</PopoverMenuItem>
-					<PopoverMenuItem icon="info-outline" onClick={ this.goToDNSManagement }>
-						{ translate( 'Manage DNS' ) }
-					</PopoverMenuItem>
+					{ ! (
+						domain.type === domainTypes.SITE_REDIRECT ||
+						domain.transferStatus === transferStatus.PENDING_ASYNC
+					) &&
+						domain.canManageDnsRecords &&
+						domain.transferStatus !== transferStatus.PENDING_ASYNC && (
```

Comment:
```suggestion
					{ ! (
						domain.type === domainTypes.SITE_REDIRECT ||
						domain.transferStatus === transferStatus.PENDING_ASYNC
					) &&
						domain.canManageDnsRecords && (
```

This can be simplified, because if `domain.transferStatus === transferStatus.PENDING_ASYNC` is true the first part of the condition will be false